### PR TITLE
feat: support selects in modals

### PR DESCRIPTION
**Pull request by onerandomusername**
## Summary

adds selects to modals and modal interactions. 

When I made this, I think I was forgetting something, but I can't remember what it was...

## Checklist

- [x] If code changes were made, then they have been tested
    - [x] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint`
    - [x] I have type-checked the code by running `task pyright`
- [ ] This PR fixes an issue
- [x] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

**Comment by onerandomusername** [comment]
Depends on #559 due to the pyright version bump.

**Comment by shiftinv** [comment]
Depends on #583

File: disnake/components.py [hunk 1]
```diff
@@ -279,6 +276,7 @@ def __init__(self, data: SelectMenuPayload):
         self.options: List[SelectOption] = [
             SelectOption.from_dict(option) for option in data.get("options", [])
         ]
+        self.values: List[str] = data.get("values", [])
```
Comment (shiftinv) [inline disnake/components.py:279]: I believe this can be removed now that #583 is merged

File: examples/modal.py [hunk 2]
```diff
@@ -8,7 +8,7 @@
 
 class MyModal(disnake.ui.Modal):
     def __init__(self) -> None:
-        components = [
+        components = disnake.ui.ModalActionRow(
```
Comment (shiftinv) [inline examples/modal.py:11]: This doesn't quite work as action rows can only contain one of these components each, we can probably keep it as a list like before

File: examples/modal.py [hunk 3]
```diff
@@ -55,34 +75,45 @@ async def create_tag(inter: disnake.CommandInteraction):
 async def create_tag_low(inter: disnake.CommandInteraction):
     # Works same as the above code but using a low level interface.
     # It's recommended to use this if you don't want to increase cache usage.
+    components = disnake.ui.ModalActionRow(
```
Comment (shiftinv) [inline examples/modal.py:78]: same thing here 🤔

File: examples/modal.py [hunk 4]
```diff
@@ -98,8 +129,15 @@ async def create_tag_low(inter: disnake.CommandInteraction):
         return
 
     embed = disnake.Embed(title="Tag Creation")
-    for custom_id, value in modal_inter.text_values.items():
-        embed.add_field(name=custom_id.capitalize(), value=value, inline=False)
+
+    # text_values is a dict of TextInput.custom_id to the inputted value.
```
Comment (shiftinv) [inline examples/modal.py:133]:
```suggestion
    # text_values is a dict of TextInput.custom_id to the entered value.
```

File: examples/modal.py [hunk 5]
```diff
@@ -98,8 +129,15 @@ async def create_tag_low(inter: disnake.CommandInteraction):
...
+    for key, value in modal_inter.text_values.items():
+        embed.add_field(name=key.capitalize(), value=value, inline=False)
+
+    # select_values is a dict of Select.custom_id to the selected value.
```
Comment (shiftinv) [inline examples/modal.py:137]:
```suggestion
    # select_values is a dict of Select.custom_id to the selected values.
```

File: disnake/ui/action_row.py [hunk 6]
```diff
@@ -269,7 +269,7 @@ def add_button(
     def add_select(
         self: Union[
             ActionRow[MessageUIComponent],
-            # ActionRow[ModalUIComponent],
+            ActionRow[ModalUIComponent],
```
Comment (shiftinv) [inline disnake/ui/action_row.py:272]: The docstring still shows `Can only be used if the action row holds message components`, we can just remove that section now, imo.

Comment (onerandomusername) [reply disnake/ui/action_row.py:272]: https://github.com/DisnakeDev/disnake/pull/558/commits/31d2e119bdf7c5597c4d3af2f41dc0e3d4f6eee3

File: disnake/components.py (hunk 1 above)
Comment (onerandomusername) [reply disnake/components.py]: https://github.com/DisnakeDev/disnake/pull/558/commits/31d2e119bdf7c5597c4d3af2f41dc0e3d4f6eee3

File: examples/modal.py (hunk 3 above)
Comment (onerandomusername) [reply examples/modal.py]: https://github.com/DisnakeDev/disnake/pull/558/commits/31d2e119bdf7c5597c4d3af2f41dc0e3d4f6eee3

File: examples/modal.py (hunk 2 above)
Comment (onerandomusername) [reply examples/modal.py]: https://github.com/DisnakeDev/disnake/pull/558/commits/31d2e119bdf7c5597c4d3af2f41dc0e3d4f6eee3

File: examples/modal.py (hunk 4 above)
Comment (onerandomusername) [reply examples/modal.py]: https://github.com/DisnakeDev/disnake/pull/558/commits/31d2e119bdf7c5597c4d3af2f41dc0e3d4f6eee3

File: examples/modal.py (hunk 5 above)
Comment (onerandomusername) [reply examples/modal.py]: https://github.com/DisnakeDev/disnake/pull/558/commits/31d2e119bdf7c5597c4d3af2f41dc0e3d4f6eee3

**Review by shiftinv — APPROVED** [review]

**Comment by onerandomusername** [comment]
@shiftinv Not sure why this was merged yet...?
image

**Pull request merged**